src/hotspot/share/cds/aotLogging.hpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_CDS_AOTLOGGING_HPP
+#define SHARE_CDS_AOTLOGGING_HPP
+
+#include "cds/cds_globals.hpp"
+#include "cds/cdsConfig.hpp"
+#include "logging/log.hpp"
+
+// UL Logging for AOT
+// ==================
+//
+// The old "CDS" feature is rebranded as "AOT" in JEP 483. Therefore, UL logging
+// related to the AOT features should be using the [aot] tag.
+//
+// However, some old scripts may be using -Xlog:cds for diagnostic purposes. To
+// provide a fair amount of backwards compatibility for such scripts, some AOT
+// logs that are likely to be used by such scripts are printed using the macros
+// in this header file.
+//
+// NOTE: MOST of the AOT logs will be using the usual macros such as log_info(aot)(...).
+// The information below does NOT apply to such logs.
+//
+// CDS Compatibility Logs & Compatibility Macros
+// =============================================
+//
+// A subset of the original CDS logs (the "CDS Compatibility Logs") have been
+// selected in JDK 25. These logs are guarded using the aot_log_xxx compatibility
+// macros. Before JDK 25, such code looked like this:
+//
+//      log_info(cds)("trying to map %s%s", info, _full_path);
+//      log_warning(cds)("Unable to read the file header.");
+//
+// New code since JDK 25:
+//
+//      aot_log_info(aot)("trying to map %s%s", info, _full_path);
+//      aot_log_warning(aot)("Unable to read the file header.");
+//
+// The messages printed with the log_aot_xxx() macros work as if they are
+// using the [cds] tag when running with the "classic" CDS flags such as
+// -XX:SharedArchiveFile:
```

Comment:
What if you specify no flags and rely on the implicit settings e.g. `-Xshare:auto`, will `-Xlog:cds` continue to work as today?

Reply:
I updated the comments to flip the condition -- if no `-XX:AOTxxx` flag are used, we print the `[cds]` decorations.

The new AOT workflow requires the use of at least one flag that starts with `-XX:AOT`. There's no implicit use of AOT features.